lib/cutting_edge/app.rb

```diff
@@ -35,18 +35,37 @@ module CuttingEdge
   class App < Sinatra::Base
     include CuttingEdgeHelpers
 
+    set :views, ::File.join(::File.dirname(__FILE__), 'templates')
+    Tilt.register Tilt::ERBTemplate, 'html.erb'
+    set :public_folder, ::File.join(::File.dirname(__FILE__), 'public')
+
     logger filename: "#{settings.environment}.log", level: :trace
 
     before do
       @store = settings.store
     end
 
-    get %r{/(.+)/(.+)/(.+)/info} do |source, org, name|
+    get "/" do
+      @repos = CuttingEdge::App.repositories
+      erb :index
+    end
+
+    get %r{/(.+)/(.+)/(.+)/info/json} do |source, org, name|
       repo_defined?(source, org, name)
       content_type :json
       @store[@repo.identifier].merge({:language => @repo.lang}).to_json # Todo: check whether value exists yet? If not, call worker / wait / timeout?
```

Comment:
Check for nil here and fail accordingly

Reply:
How should this fail? I was thinking of doing this:
```ruby
      data = @store[@repo.identifier] || {}
      data.merge({:language => @repo.lang}).to_json 
```
But that way it isn't necessarily clear that there was a failure. Or we could just `halt 500`. Thoughts?

Reply:
Implemented `status 500`.